src/pages/StocktakeEditPage.js

```diff
@@ -141,7 +140,7 @@ export class StocktakeEditPage extends React.Component {
   assignReason = stocktakeItem => {
     const { database } = this.props;
     if (stocktakeItem.shouldApplyReason) {
-      this.onOpenReasonModal();
+      this.setState({ isReasonsModalOpen: true, currentStocktakeItem: stocktakeItem });
     } else {
       stocktakeItem.applyReasonToBatches(database);
```

Comment:
I think this method call is a bit confusing, defo needs a comment or explicitly pass `null`. It looks like @joshxg removed his previous `removeReason` but I do think there should be a `clearReasonsFromBatches` as it's a lot clearer what's going on.

Reply:
I 100% agree, but I wasn't going that far refactoring it.